Source/Core/VideoBackends/Vulkan/CMakeLists.txt

```diff
@@ -28,14 +28,22 @@ set(LIBS
   common
 )
 
+add_dolphin_library(videovulkan "${SRCS}" "${LIBS}")
+
 # Only include the Vulkan headers when building the Vulkan backend
-include_directories(${CMAKE_SOURCE_DIR}/Externals/Vulkan/Include)
+target_include_directories(videovulkan PRIVATE ${CMAKE_SOURCE_DIR}/Externals/Vulkan/Include)
 
 # Silence warnings on glslang by flagging it as a system include
-include_directories(SYSTEM ${CMAKE_SOURCE_DIR}/Externals/glslang/glslang/Public)
-include_directories(SYSTEM ${CMAKE_SOURCE_DIR}/Externals/glslang/SPIRV)
+target_include_directories(videovulkan SYSTEM PRIVATE ${CMAKE_SOURCE_DIR}/Externals/glslang/glslang/Public)
+target_include_directories(videovulkan SYSTEM PRIVATE ${CMAKE_SOURCE_DIR}/Externals/glslang/SPIRV)
 
 # Link against glslang, the other necessary libraries are referenced by the executable.
-add_dolphin_library(videovulkan "${SRCS}" "${LIBS}")
 target_link_libraries(videovulkan PRIVATE glslang)
 
+# When building for X11, we need to link to libX11-xcb.
+# The main executable doesn't need this library directly, so we have to add it here.
+if(USE_X11)
+  find_package(X11_XCB REQUIRED)
+  target_link_libraries(videovulkan PRIVATE X11::XCB)
```

Comment:
You should search for the requirements first, then proceed on adding this module if they're met, then drop the REQUIRED.